bundles/edu.kit.kastel.sdq.eclipse.grading.view/src/edu/kit/kastel/eclipse/grading/view/commands/AnnotationFilter.java

```diff
@@ -0,0 +1,25 @@
+package edu.kit.kastel.eclipse.grading.view.commands;
+
+import java.util.Arrays;
+
+import org.eclipse.jface.viewers.Viewer;
+import org.eclipse.jface.viewers.ViewerFilter;
+
+import edu.kit.kastel.eclipse.common.api.model.IMistakeType;
+
+public class AnnotationFilter extends ViewerFilter {
+	private String filter;
+	
+	public void setFilterString(String filter) {
+		this.filter = filter.toLowerCase();
+	}
+	
+	@Override
+	public boolean select(Viewer viewer, Object parent, Object element) {
+		if (this.filter != null) {
+			return Arrays.stream(((IMistakeType) element).getButtonText().toLowerCase().split(" ")).anyMatch(w -> w.startsWith(this.filter));
```

Comment:
I would also include the description of the button

Reply:
I've changed the filtering logic to include the description, but tbf I don't find it helpful because it now displays way more results when only typing a few characters and at least I usually know the name of the button I'm searching for

Reply:
Then maybe just add a switch (simple constant in the code) and disable it :)

Reply:
I've added an option to the advanced preferences to include the button description